src/web/interfaces.ts

```diff
@@ -0,0 +1,114 @@
+import { Directions } from '../Directions';
+import { State } from '../State';
+
+export interface HitSlop {
+  left?: number;
+  right?: number;
+  top?: number;
+  bottom?: number;
+  horizontal?: number;
+  vertical?: number;
+  width?: number;
+  height?: number;
+}
+
+export interface Handler {
+  handlerTag: number;
+}
+
+type ConfigArgs =
+  | number
+  | boolean
+  | HitSlop
+  | Directions
+  | Handler[]
+  | null
+  | undefined;
+
+export interface Config extends Record<string, ConfigArgs> {
+  enabled?: boolean;
+  simultaneousHandlers?: [] | null;
+  waitFor?: any[] | null;
+  hitSlop?: HitSlop;
+
+  minPointers?: number;
+  maxPointers?: number;
+  minDist?: number;
+  minDistSq?: number;
+  minVelocity?: number;
+  minVelocityX?: number;
+  minVelocityY?: number;
+  minVelocitySq?: number;
+  maxDist?: number;
+  maxDistSq?: number;
+  failOffsetXStart?: number;
+  failOffsetYStart?: number;
+  failOffsetXEnd?: number;
+  failOffsetYEnd?: number;
+  activeOffsetXStart?: number;
+  activeOffsetXEnd?: number;
+  activeOffsetYStart?: number;
+  activeOffsetYEnd?: number;
+  numberOfPointers?: number;
+  minDurationMs?: number;
+  numberOfTaps?: number;
+  maxDurationMs?: number;
+  maxDelayMs?: number;
+  maxDeltaX?: number;
+  maxDeltaY?: number;
+  direction?: Directions;
+}
+
+type NativeEventArgs = number | State | boolean | undefined;
+interface NativeEvent extends Record<string, NativeEventArgs> {
+  numberOfPointers: number;
+  state: State;
+  pointerInside: boolean | undefined;
+  handlerTag: number;
+  target: number;
+  oldState?: State;
+}
+
+export interface ResultEvent extends Record<string, NativeEvent | number> {
+  nativeEvent: NativeEvent;
+  timeStamp: number;
+}
+
+export interface PropsRef {
+  onGestureHandlerEvent: () => void;
+  onGestureHandlerStateChange: () => void;
+}
+
+export interface AdaptedPointerEvent {
+  x: number;
+  y: number;
+  offsetX: number;
+  offsetY: number;
+  pointerId: number;
+  eventType: EventTypes;
+  pointerType: string;
+  buttons: number;
+  time: number;
+}
+
+export enum MouseButtons {
+  NONE,
+  LEFT,
+  RIGHT,
+  LEFT_RIGHT,
+  SCROLL,
+  SCROLL_LEFT,
+  SCROLL_RIGHT,
+  SCROLL_LEFT_RIGHT,
+}
+
+export enum EventTypes {
+  DOWN,
+  NEXT_POINTER_DOWN,
+  UP,
+  ADDITIONAL_POINTER_UP,
```

Comment:
I'd stick to one convention

Reply:
Sure, I wanted to use next because it is clear that it's another pointer, but I've lacked a counterpart for pointer up. I'll stick to additional